### PR TITLE
Make this compatible with Node.js 12 and newer

### PR DESCRIPTION
**Pull request by lubo**
Tested with `$ npx jasmine-node spec` (linting fails even on `master`, that's why `$ npm run test` wasn't used). Logs in `spec/logs` are empty except for `agent.log`, which says:

```
[11:54:31] Loading TwemProxy config
[11:54:31] Redis Sentinel TwemProxy Agent Started on: Thu Feb 17 2022 11:54:31 GMT+0100 (Central European Standard Time)
[11:54:31] Subscribing to sentinel.
```

This needs proper end-to-end testing, hence the draft.

### CODE REVIEW

**Comment by jbartak** [comment]
`yaml.safeDump` should be substituted with `yaml.dump`.
https://github.com/QualityUnit/redis-twemproxy-agent/blob/fd4b4be441523f2cfae968fd73de7cbb754ea050/lib/agent.js#L114
```
Error: Function yaml.safeDump is removed in js-yaml 4. Use yaml.dump instead, which is now safe by default.
    at Object.safeDump (/opt/nutcracker-agent/node_modules/js-yaml/index.js:10:11)
    at Agent.save_twemproxy_config (/opt/nutcracker-agent/lib/agent.js:114:50)
    at /opt/nutcracker-agent/lib/agent.js:143:33
    at /opt/nutcracker-agent/node_modules/async/lib/async.js:551:21
    at /opt/nutcracker-agent/node_modules/async/lib/async.js:227:13
    at iterate (/opt/nutcracker-agent/node_modules/async/lib/async.js:134:13)
    at async.eachSeries (/opt/nutcracker-agent/node_modules/async/lib/async.js:150:9)
    at _asyncMap (/opt/nutcracker-agent/node_modules/async/lib/async.js:226:9)
    at Object.mapSeries (/opt/nutcracker-agent/node_modules/async/lib/async.js:216:23)
    at Object.async.series (/opt/nutcracker-agent/node_modules/async/lib/async.js:549:19)
```

**Comment by jbartak** [comment]
I was able to start `nutcracker-agent` without any other error messages:
```
$ /usr/bin/node /opt/nutcracker-agent/bin/redis_twemproxy_agent
[12:18:24] Loading TwemProxy config
[12:18:24] Redis Sentinel TwemProxy Agent Started on: Thu Feb 17 2022 12:18:24 GMT+0000 (Coordinated Universal Time)
[12:18:24] Subscribing to sentinel.
[12:18:24] Connection to Redis Sentinel established.
[12:18:24] Getting latest list of masters...
[12:18:24] Master received: la_devel_visitors 172.17.1.161:6379
[12:18:24] Updating Master la_devel_visitors to 172.17.1.161:6379
[12:18:24] Saving TwemProxy config
```

The same output I can find in `/var/log/nutcracker-agent.log`
I also tested that `nutcracker-agent` is able to react on `redis-sentinel` restart by:
```
[12:16:44] Error: Connection to Redis Sentinel was closed!
[12:17:13] WARNING: Connection to Redis Sentinel has failed 10 times!
...
[12:17:23] Saving TwemProxy config
[12:17:23] TwemProxy restarted with output: 
[12:17:23] Restarting nutcracker (via systemctl):  [  OK  ]


```
so, this proves it is able to connect to it.

File: package.json [hunk 1]
```diff
@@ -24,14 +24,14 @@
...
-    "js-yaml": "~2.1.0"
+    "cli": "^1.0.1",
+    "js-yaml": "^4.1.0",
+    "redis": "~0.8.2",
+    "underscore": "^1.13.2"
   },
   "devDependencies": {
-    "jshint": "~1.0.0",
-    "jasmine-node": "~1.3.0"
+    "jasmine-node": "^1.16.0",
```
Comment (jbartak) [inline package.json:34]: Why doesn't this package use `3.0.0` but specifically this version?
```
$ npm outdated
Package       Current  Wanted  Latest  Location                   Depended by
async          0.2.10  0.2.10   3.2.3  node_modules/async         nutcracker-agent
jasmine-node   1.16.0  1.16.2   3.0.0  node_modules/jasmine-node  nutcracker-agent
redis           0.8.6   0.8.6   4.0.3  node_modules/redis         nutcracker-agent
npm notice 
npm notice New minor version of npm available! 8.1.2 -> 8.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.5.0
npm notice Run npm install -g npm@8.5.0 to update!
npm notice
```

Comment (lubo) [reply package.json:34]: First of all, minimizing breaking changes. There's simply no good reason to abandon v1 at this point. Moreover, >1.16.0 requires vulnerable dependencies.

**Review by jbartak — DISMISSED** [review]

**Review by jbartak — APPROVED** [review]

**Pull request merged**